### PR TITLE
Add Activity Type Tutorial

### PR DESCRIPTION
**Pull request by rdenarie**

### CODE REVIEW

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 1]
```diff
@@ -3,3 +3,198 @@
 > 🚧 Work in progress
 > 
 > This page will describe how to create a new activity type in eXo Platorm
+
+In this tutorial, we will see how to add a new activity type in eXo Platform.
+Activities are short messages sent by users of the platform in different spaces stream. We already have few activity types like this :
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
Activities are short messages sent by (or on belhalf of) users of the platform in different spaces stream. Several types of activity are built-in such as : 
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 2]
```diff
@@ -3,3 +3,198 @@
...
+- Poll activity
+
+  ![](/img/activity-type/poll-activity.png)
+
+- Kudo activity
+
+  ![](/img/activity-type/kudo-activity.png)
+
+
+Each activity type define how the activity object is displayed.
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
Each activity type defines how the activity object is displayed. 
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 3]
```diff
@@ -3,3 +3,198 @@
...
+  ![](/img/activity-type/poll-activity.png)
+
+- Kudo activity
+
+  ![](/img/activity-type/kudo-activity.png)
+
+
+Each activity type define how the activity object is displayed. 
+
+To illustrate this tutorial, we will create a new activity type named "Mood of the day"
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
In this tutorial, we will create a new activity type called "Mood of the day"
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 4]
```diff
@@ -3,3 +3,198 @@
...
+In this file, you can see the props activity : 
+```vue
+  props: {
+    activity: {
+      type: Object,
+      default: null,
+    },
+  },
+```
+This props comes from the upper vue component, and represent the activity object we want to display.
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
These props come from the upper Vue component, and represent the activity object to display.
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 5]
```diff
@@ -3,3 +3,198 @@
...
+We will display it with a v-card, containing two lines : the first one (the v-card title) will contains "Mood of the day", and the second one (the v-card subtitle) will contain the activity title.
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
Let's display it with a v-card, containing two lines : the first one (the v-card title) with "Mood of the day", and the second one (the v-card subtitle) with the activity title.
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 6]
```diff
@@ -3,3 +3,198 @@
...
+      }),
+      // Whether to display edit button or not when user have permission
+      canEdit: () => false,
+      canShare: () => false,
+    },
+  });
+}
+```
+
+This file allow to register our Vue component in the eXo Platform extensionRegistry. The two first parameter of the function registerExtension define which kind of extension we are registring, the third one defines the object we want to register.
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
This file allow sto register our Vue component in the eXo Platform extensionRegistry. The first two parameters of the function registerExtension define which kind of extension we are registering, the third one defines the object we want to register.
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 7]
```diff
@@ -3,3 +3,198 @@
...
+The 'type' parameter is our extension type. All activities with this type we be displayed with our Vue component.
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
The 'type' parameter is our activity type. All activities with this type we be displayed with our Vue component.
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 8]
```diff
@@ -3,3 +3,198 @@
...
+
+## Load the Vue Component
+
+Create the folder `$EXO_HOME/sources/docs-sample/activity-extensions/src/main/webapp/groovy/MoodOfTheDayActivityExtension.gtmpl`
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
Create the template file `$EXO_HOME/sources/docs-sample/activity-extensions/src/main/webapp/groovy/MoodOfTheDayActivityExtension.gtmpl`
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 9]
```diff
@@ -3,3 +3,198 @@
...
+
+Fill this file with
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
Fill it with 
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 10]
```diff
@@ -3,3 +3,198 @@
...
+          <name>templates</name>
+          <description>Add MoodOfTheDayActivityType in portal</description>
+          <value>war:/groovy/MoodOfTheDayActivityExtension.gtmpl</value>
+        </values-param>
+      </init-params>
+    </component-plugin>
+  </external-component-plugins>
+```
+
+This xml configuration will indicate to the portal, to insert the gtmpl `.MoodOfTheDayActivityExtension.gtmpl` in the dynamic container `UIPortalApplication-End-Body` when loading the portal. The gtmpl load the javascript file which load the activity extension.
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
This xml configuration tells the portal to insert the gtmpl `.MoodOfTheDayActivityExtension.gtmpl` in the dynamic container `UIPortalApplication-End-Body` when loading the portal. The gtmpl loads the javascript file which loads our activity type extension. 
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 11]
```diff
@@ -3,3 +3,198 @@
...
+  </external-component-plugins>
+```
+
+This xml configuration will indicate to the portal, to insert the gtmpl `.MoodOfTheDayActivityExtension.gtmpl` in the dynamic container `UIPortalApplication-End-Body` when loading the portal. The gtmpl load the javascript file which load the activity extension. 
+
+
+
+## Deploy the extension
+
+1.  We will now build and deploy our extension
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
1.  Let's build our extension
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 12]
```diff
@@ -3,3 +3,198 @@
...
+    ```bash
+    cd $EXO_HOME/sources/docs-sample/activity-extensions
+    mvn clean package
+    ```
+
+2. And deploy it to the server
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
2. Now deploy it to the server
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 13]
```diff
@@ -3,3 +3,198 @@
...
+    ```
+
+3.  Update docker compose configuration to mount the new war file in the webapps of the server. The **volumes** section should be like this :
+    ```
+      volumes:
+        - exo_data:/srv/exo
+        - exo_logs:/var/log/exo
+        - $EXO_HOME/webapps/activity-extensions.war:/opt/exo/webapps/activity-extensions.war
+    ```
+3.  Restart the docker containers of eXo platform :
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
3.  Restart the docker containers of eXo Platform :
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 14]
```diff
@@ -3,3 +3,198 @@
...
+    ```bash
+    docker-compose -f docker-compose.yml up
+    ```      
+
+## Create an activity with the new type
+To create an activity we will use REST service which [manage activities](/guide/openapi/social.html#/v1%2Fsocial%2Factivities/postActivity).
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
Now let's call the [social API](/guide/openapi/social.html#/v1%2Fsocial%2Factivities/postActivity) to post a new mood activity.
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 15]
```diff
@@ -3,3 +3,198 @@
...
+
+To use it, we firstly need a space in which we will create the activity. Access to the application, and create a new space. Then access to url to get user spaces :
+
+    http://{eXoServer}/portal/rest/v1/social/spaces?limit=10
+
+Find your space and note the attribute "id". If the space is the first one create, the id will be 1
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
Find your space and note the attribute "id". If the space is the first one created, the id will be 1
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 16]
```diff
@@ -3,3 +3,198 @@
...
+
+The create the activity, we need to make a POST request. In this example, we will make it with curl, but you can use tools like [Talent API Tester](https://chrome.google.com/webstore/detail/talend-api-tester-free-ed/aejoelaoggembcahagimdiliamlcdmfm?hl=fr) or [YARC](chrome-extension://ehafadccdcdedbhcbddihehiodgcddpl/index.html) which will help to make REST requests.
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
To generate the activity, we need to make a POST request. In this example, we will make it with curl, but you can use tools like [Talend API Tester](https://chrome.google.com/webstore/detail/talend-api-tester-free-ed/aejoelaoggembcahagimdiliamlcdmfm?hl=fr) or [YARC](chrome-extension://ehafadccdcdedbhcbddihehiodgcddpl/index.html) which will help to make REST requests.
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 17]
```diff
@@ -3,3 +3,198 @@
...
+
+With curl tool, launch theses requests in your command line tool :
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
With curl tool, make the following requests from command line  : 
```

File: docs/guide/developer-guide/simple-integration/activity-type.md [hunk 18]
```diff
@@ -3,3 +3,198 @@
...
+The create the activity, we need to make a POST request. In this example, we will make it with curl, but you can use tools like [Talent API Tester](https://chrome.google.com/webstore/detail/talend-api-tester-free-ed/aejoelaoggembcahagimdiliamlcdmfm?hl=fr) or [YARC](chrome-extension://ehafadccdcdedbhcbddihehiodgcddpl/index.html) which will help to make REST requests.
+
+With curl tool, launch theses requests in your command line tool : 
+
+```bash
+curl -X POST -c cookies -d 'username=root&password=password' http://localhost:8080/portal/login
+curl -X POST -b cookies -H 'Content-Type: application/json' -d '{"title": "Im happy","type": "moodActivity"}' http://localhost:8080/portal/rest/v1/social/activities?spaceId=1
+```
+
+The first command make the login and store cookies in `cookies` file. The second command makes the POST request using `cookies file for the authentication.
```
Comment (plamarque) [inline docs/guide/developer-guide/simple-integration/activity-type.md]:
```suggestion
The first command performs the authentication and store cookies in `cookies` file. The second command makes the POST request using `cookies` file for the authentication. 
```

**Review by plamarque — COMMENTED** [review]
typos and language suggestions

**Review by ahamdi — APPROVED** [review]

**Comment by rdenarie** [comment]
Thanks @plamarque for the review :)

**Review by ahamdi — APPROVED** [review]

**Pull request merged**